x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceServiceTests.java

```diff
@@ -166,9 +168,9 @@ private void assertThatBothTasksAreTriggered(Answer<?> deleteExpiredDataAnswer,
             latch.await(5, TimeUnit.SECONDS);
         }
 
-        verify(client, Mockito.atLeast(2)).threadPool();
-        verify(client, Mockito.atLeast(1)).execute(same(DeleteExpiredDataAction.INSTANCE), any(), any());
-        verify(client, Mockito.atLeast(1)).execute(same(GetJobsAction.INSTANCE), any(), any());
+        verify(client, times(3)).threadPool();
+        verify(client, times(1)).execute(same(DeleteExpiredDataAction.INSTANCE), any(), any());
+        verify(client, times(2)).execute(same(GetJobsAction.INSTANCE), any(), any());
```

Comment:
The reason these lines used to have `atLeast` was because of the changes of #80000, which was fixing the failure of #79971. Are you sure you've fixed that in a better way? If there's still a risk of the race condition it would be best to continue to use `atLeast`.

Reply:
If that fix is needed, shouldn't the same fix be applied to the other tests, such as 
`testJobInDeletingStateAlreadyHasDeletionTask`? Looks like that's was forgotten. I'm happy to apply it everywhere though.

Reply:
@davidkyle Could you chime in on this? Thanks!

Reply:
It does seem wrong as the logic here that supplies the schedule for the service to run should stop mean that further calls don't happen. In theory it should not have to be `atleast` because the countdown latch restricts the number of calls and each call to the scheduler is a single shot. It is not a repeated timer that goes off every 100ms. 

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceServiceTests.java#L278

I think the problem is in the logic of the schedule provider lambda and it should but this:
```
            if (latch.getCount() > 0) {
                latch.countDown();
            }
            if (latch.getCount() > 0) {
                return TimeValue.timeValueMillis(100);
            } else {
                return TimeValue.timeValueHours(1);
            }
```

Reply:
Yes, I agree.

And the following is even simpler:
```
            latch.countDown();
            if (latch.getCount() > 0) {
                return TimeValue.timeValueMillis(100);
            } else {
                return TimeValue.timeValueHours(1);
            }
```

Reply:
The `return TimeValue.timeValueHours(1);` is never reached under normal circumstances. And if it is, the test fails with one trigger too many.

Reply:
This is fixed now. @davidkyle PTAL